internal/client/tanzupackagerepository/packagerepositoryavailability/availability.go

```diff
@@ -0,0 +1,44 @@
+/*
+Copyright © 2023 VMware, Inc. All Rights Reserved.
+SPDX-License-Identifier: MPL-2.0
+*/
+
+package tanzupackagerepositoryavailabilityclient
+
+import (
+	"github.com/vmware/terraform-provider-tanzu-mission-control/internal/client/transport"
+	"github.com/vmware/terraform-provider-tanzu-mission-control/internal/helper"
+	pkgrepository "github.com/vmware/terraform-provider-tanzu-mission-control/internal/models/tanzupackagerepository"
+)
+
+const (
+	apiVersionAndGroup = "v1alpha1/clusters"
+	apiKind            = "tanzupackage/repositories:setavailability"
+	namespaces         = "namespaces"
+)
+
+// New creates a new repository resource service API client.
+func New(transport *transport.Client) ClientService {
+	return &Client{Client: transport}
+}
+
+/*
+Client for repository resource service API.
+*/
+type Client struct {
+	*transport.Client
+}
+
+// ClientService is the interface for Client methods.
+type ClientService interface {
+	SetRepositoryAvailability(request *pkgrepository.VmwareTanzuManageV1alpha1ClusterNamespaceTanzupackageRepositorySetRepositoryAvailabilityRequest) (*pkgrepository.VmwareTanzuManageV1alpha1ClusterNamespaceTanzupackageRepositorySetRepositoryAvailabilityResponse, error)
```

Comment:
This is a separate service? And not included as part of of UPDATE API of `RepositoryResourceService`?

Reply:
yes this is a separate service just to change the availability of the repository and it's not needed to be included as a part of UPDATE API of RepositoryResourceService

I am using this as part of the Update operational handlers functions for the package repository.